src/java/com/sarality/app/view/list/NonScrollingListViewInitializer.java

```diff
@@ -25,19 +25,30 @@
 public class NonScrollingListViewInitializer<T> extends BaseViewInitializer<LinearLayout, List<T>> {
 
   private final ListViewRowRenderer<T> rowRenderer;
+  private final ListViewRowRenderer<T> emptyViewRenderer;
 
   public NonScrollingListViewInitializer(FragmentActivity activity, LinearLayout listView,
-      ListViewRowRenderer<T> rowRenderer) {
+                                         ListViewRowRenderer<T> rowRenderer, ListViewRowRenderer<T> emptyViewRenderer) {
     super(activity, listView);
     this.rowRenderer = rowRenderer;
+    this.emptyViewRenderer = emptyViewRenderer;
   }
 
   @Override
   public void render(List<T> dataList) {
+    getView().removeAllViews();
+
+    if (dataList.size() == 0) {
+      LayoutInflater inflater = LayoutInflater.from(getContext());
+      View rowView = inflater.inflate(emptyViewRenderer.getRowLayout(null), getView());
```

Comment:
Don't inflate it. Use the same paradigm as ListViewInitializer where the layout must include the View for the empty view.